cocoa/tests_backend/window.py

```diff
@@ -27,16 +30,34 @@ async def wait_for_window(
         message,
         minimize=False,
         full_screen=False,
-        state_switch_not_from_normal=False,
+        expected_state=None,
     ):
-        await self.redraw(
-            message,
-            delay=(
-                1.75
-                if state_switch_not_from_normal
-                else 0.75 if full_screen else 0.5 if minimize else 0.1
-            ),
-        )
+        await self.redraw(message, delay=0.1)
+        if expected_state:
+            timeout = 5
+            polling_interval = 0.1
+            exception = None
+            loop = asyncio.get_running_loop()
+            start_time = loop.time()
+            while (loop.time() - start_time) < timeout:
+                try:
+                    assert self.instantaneous_state == expected_state
+                    assert self.window._impl._pending_state_transition is None
+                    return
+                except AssertionError as e:
+                    exception = e
+                    await asyncio.sleep(polling_interval)
+                    continue
+                raise exception
+
+    async def wait_for_window_close(self, pre_close_window_state):
```

Comment:
Can't `pre_close_window_state` be determined from the window itself?

Reply:
Once, `window.close()` is called, then determining the window state would be unreliable. To avoid passing the parameter, we could determine the pre_close_window_state and close the window at the window probe.

Reply:
Should I change wait_for_window_close() to close_and_wait_for_window(), so that we could determine the pre_close_window_state and close the window at the window probe?

Reply:
When you start proposing method names like `close_and_wait_for_window()`, you should be stepping back and asking "is there a better name for this?". In this case - it's literally a cleanup method.. so... `window_probe.cleanup()` (doing both the close and wait) would make sense to me.

Reply:
On the subject of naming, `expected_state` is also fairly verbose - unless you're expecting a different state to be passed into the window, `state` is fine as an argument name.

Reply:
I have added a `window_probe.cleanup()` method and replaced `expected_state` with `state`.

Reply:
I have added a `window_probe.cleanup()` method and replaced `expected_state` with `state`.